coderd/templateversions.go

```diff
@@ -640,6 +640,68 @@ func (api *API) templateVersionByOrganizationAndName(rw http.ResponseWriter, r *
 	httpapi.Write(ctx, rw, http.StatusOK, convertTemplateVersion(templateVersion, convertProvisionerJob(job), user))
 }
 
+func (api *API) previousTemplateVersionByOrganizationAndName(rw http.ResponseWriter, r *http.Request) {
+	ctx := r.Context()
+	organization := httpmw.OrganizationParam(r)
+	templateVersionName := chi.URLParam(r, "templateversionname")
+	templateVersion, err := api.Database.GetTemplateVersionByOrganizationAndName(ctx, database.GetTemplateVersionByOrganizationAndNameParams{
+		OrganizationID: organization.ID,
+		Name:           templateVersionName,
+	})
+	if errors.Is(err, sql.ErrNoRows) {
```

Comment:
I believe our linter should catch this, but maybe not? `¯\_(ツ)_/¯`

```suggestion
	if xerrors.Is(err, sql.ErrNoRows) {
```

Reply:
Also, a second suggestion: You could move this check inside `if err != nil`.

The motivation behind this is so that the error is handled in one section. It's easy for someone to accidentally put logic between this error check and the next one (`if err != nil`). It's also easier for the eye to catch what is relevant for the error handling.

(Same below.)

Reply:
What is the diff between `errors` and `xerrors`?

Reply:
We use `xerrors` because it has stack traces, and to keep things simple, we decided to only use one library for errors. For now, that's `xerrors`, but we may change this soon #5288. For this PR, though, we can stick with `xerrors` all the way.